### PR TITLE
[system-probe] Read udp ports bindings for all namespaces on init

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Add code to read UDP port bindings for all network namespaces on the host on startup of the system-probe.

### Motivation

Port binding information is used to determine connection direction in network flows; previously we were not collecting this information for all namespaces for UDP.

### Describe your test plan
This test requires `docker`, `jq`, `dig` on a linux vm/machine.

There are two tests, one where system-probe is run after the dnsmasq container is run, and one where system-probe is run before the dnsmasq container is run.

First test:
1. Run the following container `docker run --name dnsmasq -d -p 5053:53/tcp -p 5053:53/udp andyshinn/dnsmasq`
2. Run the system-probe, and either put it in the background or switch to a new prompt on the same machine.
3. Run `dig @127.0.0.1 -p 5053 +tcp google.com`
4. Run `curl --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/connections  | jq '.conns[] | select(.type=="tcp" and ((.laddr.port==53 and .direction=="incoming") or (.direction=="outgoing" and .raddr.port==5053)))'`
5. You should see two json objects output, one of them should have the `direction` field set to `outgoing` and the other one `incoming`.
6. Run `dig @127.0.0.1 -p 5053 google.com`
7. Run `curl --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/connections  | jq '.conns[] | select(.type=="udp" and ((.laddr.port==53 and .direction=="incoming") or (.direction=="outgoing" and .raddr.port==5053)))'`
8. You should see two json objects output, one of them should have the `direction` field set to `outgoing` and the other one `incoming`.
9. Stop the system-probe.
10. Stop the container `docker rm -f dnsmasq`

Second test:
1. Run the system-probe, and either put it in the background or switch to a new prompt on the same machine.
2. Run the following container `docker run --name dnsmasq -d -p 5053:53/tcp -p 5053:53/udp andyshinn/dnsmasq`
3. Run `dig @127.0.0.1 -p 5053 +tcp google.com`
4. Run `curl --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/connections  | jq '.conns[] | select(.type=="tcp" and ((.laddr.port==53 and .direction=="incoming") or (.direction=="outgoing" and .raddr.port==5053)))'`
5. You should see two json objects output, one of them should have the `direction` field set to `outgoing` and the other one `incoming`.
6. Run `dig @127.0.0.1 -p 5053 google.com`
7. Run `curl --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/connections  | jq '.conns[] | select(.type=="udp" and ((.laddr.port==53 and .direction=="incoming") or (.direction=="outgoing" and .raddr.port==5053)))'`
8. You should see two json objects output, one of them should have the `direction` field set to `outgoing` and the other one `incoming`.
9. Stop the system-probe.
10. Stop the container `docker rm -f dnsmasq`

### CODE REVIEW

File: pkg/ebpf/tracer.go [hunk 1]
```diff
@@ -206,11 +206,7 @@ func NewTracer(config *Config) (*Tracer, error) {
 	portMapping := network.NewPortMapping(config.ProcRoot, config.CollectTCPConns, config.CollectIPv6Conns)
 	udpPortMapping := network.NewPortMapping(config.ProcRoot, config.CollectTCPConns, config.CollectIPv6Conns)
 	if err := portMapping.ReadInitialState(); err != nil {
-		return nil, fmt.Errorf("failed to read initial TCP pid->port mapping: %s", err)
-	}
-
-	if err := udpPortMapping.ReadInitialUDPState(); err != nil {
-		return nil, fmt.Errorf("failed to read initial UDP pid->port mapping: %s", err)
+		return nil, fmt.Errorf("failed to read initial pid->port mapping: %s", err)
```
Comment (leeavital) [inline pkg/ebpf/tracer.go:209]: Is portMapping now meant to hold listening ports for UDP and TCP ?  If so we should remove the `udpPortMapping` field altogether.

I don't think storing udp and tcp portmappings together is right though, there can be listeners on the same port with different transports.

**Review by leeavital — COMMENTED** [review]
Please add QA instructions to the commit message.

File: pkg/ebpf/tracer.go (hunk 1 above)
Comment (hmahmood) [reply pkg/ebpf/tracer.go:209]: No I don't think this is correct. Let me fix it.

**Comment by hmahmood** [comment]
> Please add QA instructions to the commit message.

@leeavital Test instructions added.

**Review by leeavital — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
lgtm

**Pull request merged**